### DISCUSSION TITLE
Add Consumer Key and Consumer Secret as configuration values

### DISCUSSION THREAD

**Discussion by davescruggs** [forcedotcom/cli#2725] [Ideas]
Request: allow the consumer key and consumer secret to be stored in configuration values so they don't have to be passed as arguments. 

It's a best practice to create a connected app for Salesforce DX and CLI authentication: https://developer.salesforce.com/docs/atlas.en-us.sfdx_dev.meta/sfdx_dev/sfdx_dev_auth_connected_app.htm

The consumer key and consumer secret are inconvenient to use with the CLI, difficult to use in CI/CD applications (although that's typically JWT), and aren't easy (possible?) to pass through to the VS Code plugins. Consumer secrets are passed as prompts. 

If the CLI could default to using config values, then the settings could be done once, and passed through to the VS Code plugins.

**Comment by mshanemc** [comment]
Can you describe what command(s) would need this?  Is this for JWT ? 

The `org login sfdx-url` already supports the key/secret in the URL format with both "read from a file" and "read from stdin" options and that's a good CI option.

**Reply by davescruggs** [reply]
Not for JWT 

For airgapped environments, it would mean that you can use `org login web` without having to pass the consumer key on the command line or having to paste the consumer secret on a prompt. 

For VS Code, it means that you could authorize an org or a dev hub from the command palette since they'd already have been set. 

If you've created a custom connected app for dx authentication, then wouldn't you want that key and that secret to be the default values for your CLI operations? 

----

The sfdx-url is a nice approach, but can you leave out the session ID and have it work?

**Reply by mshanemc** [reply]
the sfdx-url needs an AccessToken to [pass regex](https://github.com/forcedotcom/sfdx-core/blob/dddc9a2e4d202eb481bcb9aff142850378fa6240/src/org/authInfo.ts#L364-L365).  `/^force:\/\/([a-zA-Z0-9._-]+={0,2}):([a-zA-Z0-9._-]*={0,2}):([a-zA-Z0-9._-]+={0,2})@([a-zA-Z0-9._-]+)/`

It can be expired (the RT will get you a new AT).  I'm not sure if it can be "accessToken that was never valid but fit the pattern" but you could give that a shot.
